lib/api/include/irods/generalUpdate.h

```diff
@@ -1,12 +1,19 @@
-#ifndef GENERAL_UPDATE_H__
-#define GENERAL_UPDATE_H__
+#ifndef IRODS_GENERAL_UPDATE_H
+#define IRODS_GENERAL_UPDATE_H
 
 #include "irods/rcConnect.h"
 #include "irods/rodsGeneralUpdate.h"
 
 #ifdef __cplusplus
-extern "C"
+extern "C" {
 #endif
-int rcGeneralUpdate( rcComm_t *conn, generalUpdateInp_t *generalUpdateInp );
 
+int rcGeneralUpdate(rcComm_t* conn, generalUpdateInp_t* generalUpdateInp)
+    __attribute__((deprecated("rcGeneralUpdate is deprecated and will be removed in a later release. If you are "
+                              "reaching for this, consider revising your approach.")));
```

Comment:
First of all, this formatting lol

Secondly, I feel like the wording could be a little more helpful. Is there any way to offer alternatives, or is this too "general"? Even just suggesting that other endpoints likely exist that will meet the desired use case could be helpful, I think.

Reply:
> First of all, this formatting lol

I originally had the attribute above the function prototype, but clang-format wanted to format it in a weird way. So, here we are :-)

> Secondly, I feel like the wording could be a little more helpful. Is there any way to offer alternatives, or is this too "general"? Even just suggesting that other endpoints likely exist that will meet the desired use case could be helpful, I think.

Based on the issue, this API should not exist because it allows users to change data in the database directly. Suggesting an alternative would depend on what the caller was attempting to do. For that reason, I don't think we can offer good alternatives.

Reply:
Fair enough. I felt like the message reads a little bit strangely, but maybe it's just me. I'll just resolve this.

Reply:
If you feel the wording isn't good enough, I'm happy to hear alternatives.

I could remove the second statement or remove the message entirely. There's nothing stating we must give devs a reason for the deprecation. And this is a special case given there isn't an alternative API to point devs to.

Reopening conversation so I don't forget to revisit this.

Reply:
i think it's okay.   no good hints to give.  and this text goes away in 5.0.

Reply:
Here are some last minute alternatives:
- rcGeneralUpdate is deprecated. Its use is highly discouraged.
- rcGeneralUpdate is deprecated. Please revise any code using this API.
- rcGeneralUpdate is deprecated. Its use should be avoided.

Thoughts?

Reply:
I like "rcGeneralUpdate is deprecated. Its use should be avoided."

Reply:
Agreed - we don't necessarily know the clients that are using it... and that's what deprecation means most closely... avoid.

Reply:
Will use that.

Reply:
Changed to `Its use should be avoided`. May change it to `Use of this API should be avoided.` since that reads well in the code and in the compiler output.

The message about the API being deprecated is produced by the compiler so I removed that.

Reply:
Leaving as is. The deprecation message is designed for the compiler and consumers of the API, rather than people reading the code.